### PR TITLE
Fix: Add alert for route tracker licensing error

### PR DESCRIPTION
**Pull request by yo1995**
## Background

`common-samples/issues/2288`

User may use the route tracker sample code in the sample viewer. There are a few scenarios

- User is in developer mode without a license: the basemap will show the watermark, but all functionalities would work
- User uses a Advanced license string for deployment: all good
- User uses a Lite license string: in this case, since the route tracking is not supported by Lite license, the `AGSRouteTracker.trackLocation(_:completion:)` call would fail and give error. 
    However in our (my 🤕 ) previous implementations, we didn't handle that error and let the sample fail silently. That is due to my negligence of others using Lite license to build the app. See more in #980 

Better depict with a table

||`v.next`|this branch|
|-|-|-|
|Dev mode|✅ no error|✅ no error|
|Lite|❌ error not shown, silently fail|✅ Error handled with prompt|
|Advanced|✅ no error|✅ no error|

## Fix

Add error handling and comment in the code: By nilifying the route tracker, we can stop the route tracking.

This is not an elegant way to stop route tracker from giving a series of errors while location updates keep firing, but it works. Possible ways include:
    
- ✅  Nilify the route tracker object
- Nililfy the `locationChangeHandlerDelegate` on the data source, hence stop location updates.
- When error encountered, call the `reset` method in the VC, so that no more location updates are given. All 3 samples can be reset.

Also, we can show an alert instead of updating the status label to show this error. What do you think? (like this 👇 )

error

## How to test

- Swap the Advanced license key in the dev `.secrets` file with a Lite one (which can be acquired [here](https://developers.arcgis.com/ios/license-and-deployment/)), and compare `v.next` with this branch. The former will fail silently while the latter show the error in status label.

### CODE REVIEW

File: arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift [hunk 1]
```diff
@@ -249,7 +249,15 @@ extension NavigateARNavigatorViewController: AGSLocationChangeHandlerDelegate {
     func locationDataSource(_ locationDataSource: AGSLocationDataSource, locationDidChange location: AGSLocation) {
         startButtonItem.isEnabled = true
         // Send location updates to route tracker if actively navigating.
-        routeTracker?.trackLocation(location)
+        routeTracker?.trackLocation(location)  { [weak self] error in
```
Comment (philium) [inline arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift:252]:
```suggestion
        routeTracker?.trackLocation(location) { [weak self] error in
```

**Review by philium — CHANGES_REQUESTED** [review]
Just one little tiny change requested.

File: arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift (hunk 1 above)
Comment (yo1995) [reply arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift:252]: My bad here 😅 . What do you think for nilifying `locationChangeHandlerDelegate` vs `routeTracker` in this case?

Comment (philium) [reply arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift:252]: I don't really know which is the better approach.

**Review by philium — APPROVED** [review]

**Review by vquach2404 — APPROVED** [review]

**Pull request merged**